client/src/components/Header.tsx

```diff
@@ -30,16 +31,17 @@ const Header = () => {
   };
 
   return (
-    <Tabs>
+    <Tabs index={game.level}>
       <Flex bg="gray.200" w="100%" position="sticky" zIndex="1" p={4}>
         <TabList>
           <Tab onClick={() => game.jumpToLevel(0)}>Level 1</Tab>
-          <Tab onClick={() => game.jumpToLevel(1)}>Level 2</Tab>
-          <Tab onClick={() => game.jumpToLevel(2)}>Level 3</Tab>
-          <Tab onClick={() => game.jumpToLevel(3)}>Level 4</Tab>
-          <Tab onClick={() => game.jumpToLevel(4)}>Level 5</Tab>
+          <Tab isDisabled={game.hasSeenLevel(0)} onClick={() => game.jumpToLevel(1)}>Level 2</Tab>
```

Comment:
Semantically, it doesn't make sense for the tab to be `isDisabled` when the user has indeed seen the previous level. I would change the function name to `hasNotSeenLevel`.

Reply:
Actually, using `Not` in a function name is usually indicative that your logic is flipped (you'll have a double negative when you need the `true` case). Keep the name as `hasSeenLevel` but refactor the function to be semantically correct and then make this `isDisabled={!game.hasSeenLevel(0)}`.